File: leetcode/valid-anagram.java
// https://leetcode.com/problems/valid-anagram

class Solution {
    public boolean isAnagram(String s, String t) {
        char[] one = s.toCharArray();
		char[] two = t.toCharArray();
		Arrays.sort(one);
		Arrays.sort(two);
		return String.valueOf(one).equals(String.valueOf(two));

    }
}

File: leetcode/valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.size() != t.size()) return false;
        int cnt[127]{};
        for (int i = 0; i < s.size(); ++i) {
            ++cnt[s[i]];
            --cnt[t[i]];
        }
        for (int i = 0; i < 127; ++i)
            if (cnt[i] != 0) return false;
        return true;
    }
};